### PR TITLE
Add displayName to nested render warnings [#1726]

### LINKED ISSUE #1726: Add displayName to nested render warning
We should have a more informative error message here. See https://github.com/facebook/react/pull/1402#issuecomment-46728224.

### PR DESCRIPTION
**Pull request by benmoss**
Trying to fix #1726

### CODE REVIEW

File: src/core/__tests__/ReactCompositeComponent-test.js [hunk 1]
```diff
@@ -826,7 +826,7 @@ describe('ReactCompositeComponent', function() {
     ReactTestUtils.renderIntoDocument(<Outer />);
     expect(console.warn.argsForCall.length).toBe(1);
     expect(console.warn.argsForCall[0][0]).toBe(
-      'Warning: _renderNewRootComponent(): Render methods should ' +
+      'Warning: ReactCompositeComponent._renderNewRootComponent(): Render methods should ' +
```
Comment (zpao) [inline src/core/__tests__/ReactCompositeComponent-test.js]: This should be picking up `Outer` as the display name (the transform adds it).

File: src/browser/ui/ReactMount.js [hunk 2]
```diff
@@ -518,10 +519,11 @@ var ReactMount = {
...
       ReactCurrentOwner.current == null,
-      'unmountComponentAtNode(): Render methods should be a pure function of ' +
-      'props and state; triggering nested component updates from render is ' +
-      'not allowed. If necessary, trigger nested updates in ' +
-      'componentDidUpdate.'
+      '%s.unmountComponentAtNode(): Render methods should be a pure function ' +
+      'of props and state; triggering nested component updates from render ' +
+      'is not allowed. If necessary, trigger nested updates in ' +
+      'componentDidUpdate.',
+      this.constructor.displayName || 'ReactCompositeComponent'
```
Comment (zpao) [inline src/browser/ui/ReactMount.js]: Actually, this message doesn't make sense. You'll only ever call `React.unmountComponentAtNode`, never a component itself. I think we actually want to leave the beginning of this warning in place and do something like "See the render method of %s." at the end.

Also, `this` isn't right in either of these cases. I believe `this` will refer to the global scope of the ReactMount module, not `this` inside of another component. I think you'll actually want to make use of `ReactCurrentOwner.current`.

**Comment by benmoss** [comment]
Apologies for the bad PR, I didn't fully understand the context. Hopefully this fixes those issues you caught.

**Comment by spicyj** [comment]
This looks good – will merge after 0.13.

**Pull request merged**